src/validators.ts

```diff
@@ -41,14 +57,14 @@ const archMap = {
   // a.k.a. amd64
   x64: {
     aix: "",
-    darwin: "darwin",
-    freebsd: "linux",
-    linux: "linux",
-    openbsd: "linux",
-    netbsd: "linux",
+    darwin: "darwin-amd64",
+    freebsd: "linux-amd64",
+    linux: "linux-amd64",
+    openbsd: "linux-amd64",
+    netbsd: "linux-amd64",
     sunos: "",
-    win32: "windows",
-    cygwin: "windows",
```

Comment:
I had mistakenly thought cygwin was windows, it's not... https://www.cygwin.com/
Future work might involve exploring if this can all work on cygwin somehow.